examples/differential_correction_coupled.py

```diff
@@ -0,0 +1,237 @@
+import sys
+sys.path.insert(0, './src/')
+import differential_correction as diffcorr
+import coupled_quartic_hamiltonian as coupled
+# This needs testing for installation 
+
+#%% Setting up parameters for the method and the Hamiltonian system
+
+save_final_plot = True
+show_final_plot = False
+N = 4               # dimension of phase space
+omega=1.00
+EPSILON_S = 0.0     # Energy of the saddle
+alpha = 1.00
+beta = 1.00
+epsilon= 1e-1
+parameters = np.array([1, omega, EPSILON_S, alpha, beta, omega, epsilon])
+eqNum = 1
+#deltaE_vals = [0.01, 0.1, 1.00, 2.0, 4.0]
+#linecolor = ['b','r','g','m','c']
+deltaE_vals = [0.1, 1.00]
+linecolor = ['b','r']
+
+
+eqPt = diffcorr.get_eq_pts(eqNum, coupled.init_guess_eqpt_coupled, \
+                            coupled.grad_pot_coupled, parameters)
+
+#energy of the saddle eq pt
+eSaddle = diffcorr.get_total_energy([eqPt[0],eqPt[1],0,0], \
+                                    coupled.pot_energy_coupled, parameters) 
+#If orbit is an n-array, e.g. orbit = [orbit_0, orbit_1, ..., orbit_n]
+# 
+#e = np.zeros(n,1)    
+#for i in range(n):
+#    e[i] = get_total_energy_deleonberne(orbit[i], parameters)
+#e = np.mean(e)
+
+#%%
+nFam = 100 # use nFam = 10 for low energy
+
+# first two amplitudes for continuation procedure to get p.o. family
+Ax1  = 2.e-5 # initial amplitude (1 of 2) values to use: 2.e-3
+Ax2  = 2*Ax1 # initial amplitude (2 of 2)
+
+t = time.time()
+
+#  get the initial conditions and periods for a family of periodic orbits
+po_fam_file = open("x0_diffcorr_fam_eqPt%s_coupled.dat" %eqNum,'a+')
+[po_x0Fam,po_tpFam] = diffcorr.get_POFam(eqNum, Ax1, Ax2, nFam, \
+                                        po_fam_file, \
+                                        coupled.init_guess_eqpt_coupled, \
+                                        coupled.grad_pot_coupled, \
+                                        coupled.jacobian_coupled, \
+                                        coupled.guess_lin_coupled, \
+                                        coupled.diffcorr_setup_coupled, \
+                                        coupled.conv_coord_coupled, \
+                                        coupled.diffcorr_acc_corr_coupled, \
+                                        coupled.ham2dof_coupled, \
+                                        coupled.half_period_coupled, \
+                                        coupled.pot_energy_coupled, \
+                                        coupled.varEqns_coupled, \
+                                        coupled.plot_iter_orbit_coupled, \
+                                        parameters)  
+
+poFamRuntime = time.time() - t
+x0podata = np.concatenate((po_x0Fam, po_tpFam),axis=1)
+po_fam_file.close()
+
+
+
+#%%
+# begins with a family of periodic orbits and steps until crossing the
+# initial condition with target energy 
+
+for i in range(len(deltaE_vals)):
+    deltaE = deltaE_vals[i]
+    
+    po_fam_file = open("x0_diffcorr_fam_eqPt%s_coupled.dat" %eqNum ,'a+')
```

Comment:
`with open` again

Reply:
and more below, won't make comments for each individually

Reply:
Done.